tests/Unit/Adapter/Module/Repository/ModuleRepositoryTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Adapter\Module\Repository;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Module\Repository\ModuleRepository;
+
+class ModuleRepositoryTest extends TestCase
+{
+    /**
+     * @var ModuleRepository
+     */
+    private $moduleRepository;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->moduleRepository = new ModuleRepository();
+    }
+
+    public function testNativeModulesMandatoryModules(): void
+    {
+        $modules = $this->moduleRepository->getNativeModules();
+        foreach (ModuleRepository::ADDITIONAL_ALLOWED_MODULES as $mandatoryModule) {
+            self::assertContains($mandatoryModule, $modules);
+        }
+    }
+
+    /**
+     * @dataProvider dataProviderNativeModules
+     *
+     * @param string $moduleName
+     * @param bool $isNative
+     */
+    public function testNativeModulesCheckModules(string $moduleName, bool $isNative): void
+    {
+        if ($isNative) {
+            self::assertContains($moduleName, $this->moduleRepository->getNativeModules());
+        } else {
+            self::assertNotContains($moduleName, $this->moduleRepository->getNativeModules());
+        }
+    }
+
+    public function dataProviderNativeModules(): iterable
+    {
+        // Native modules
+        yield ['blockwishlist', true];
```

Comment:
No need to test all modules, we are not testing our `composer.lock` file, but the ModuleRepositoryTest